src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs

```diff
@@ -75,7 +81,7 @@ public override async Task WriteAsync(ScriptSecretsType type, string functionNam
             List<Task> deleteTasks = new List<Task>();
             string prefix = (type == ScriptSecretsType.Host) ? HostPrefix : FunctionPrefix + Normalize(functionName);
 
-            foreach (SecretProperties item in await FindSecrets(secretsPages, x => x.Name.StartsWith(prefix)))
+            foreach (SecretProperties item in await FindSecrets(secretsPages, x => x.Name.StartsWith(prefix, StringComparison.OrdinalIgnoreCase)))
             {
                 // Delete only keys which no longer exist in passed-in secrets
                 if (!dictionary.Keys.Contains(item.Name))
```

Comment:
Can you add a log message right below here telling us that we're deleting a key, and which one it is? That would help in the future identify exactly when we are deleting customer's keys.

Reply:
And same goes for when we Set the keys below -- it'd be a huge help with debugging.